recipes/full_finetune_distributed.py

```diff
@@ -888,6 +892,18 @@ def train(self) -> None:
 
                 utils.batch_to_device(batch, self._device)
 
+                # Define optional context manager for context parallelism
+                model_inputs = list(batch.values())
+                buffers = list(self._model.buffers())
+                optional_context_parallel_context_manager = (
```

Comment:
Is this the naming we're using for other optional ctx managers? We have "activations_handling_ctx", though I'd prefer to consolidate on something like "context_parallel" or "maybe_context_parallel", I think the "with" statement says it's a context manager.

Reply:
Yeah I'm good taking out the "optional" here and matching what we do for activation offloading